File: src/_1ProgrammingBasicsWithJavaMay2023/_3ConditionalStatementsAdvanced/_3MoreExercise/_07SchoolCamp.java
package _1ProgrammingBasicsWithJavaMay2023._3ConditionalStatementsAdvanced._3MoreExercise;

import java.util.Scanner;

public class _07SchoolCamp {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String season = scanner.nextLine();
        String typeOfGroup = scanner.nextLine();
        int numberOfStudents = Integer.parseInt(scanner.nextLine());
        int numberOfStay = Integer.parseInt(scanner.nextLine());
        double pricePerNight = 0.00;
        double priceForWholeStay = 0.00;
        String typeOfSport = "";
        if (season.equals("Winter")) {
            if (typeOfGroup.equals("boys")) {
                pricePerNight = numberOfStudents * 9.60;
                priceForWholeStay = pricePerNight * numberOfStay;
                typeOfSport = "Judo";
            } else if (typeOfGroup.equals("girls")) {
                pricePerNight = numberOfStudents * 9.60;
                priceForWholeStay = pricePerNight * numberOfStay;
                typeOfSport = "Gymnastics";
            } else if (typeOfGroup.equals("mixed")) {
                pricePerNight = numberOfStudents * 10.00;
                priceForWholeStay = pricePerNight * numberOfStay;
                typeOfSport = "Ski";
            }
        } else if (season.equals("Spring")) {
            if (typeOfGroup.equals("boys")) {
                pricePerNight = numberOfStudents * 7.20;
                priceForWholeStay = pricePerNight * numberOfStay;
                typeOfSport = "Tennis";
            } else if (typeOfGroup.equals("girls")) {
                pricePerNight = numberOfStudents * 7.20;
                priceForWholeStay = pricePerNight * numberOfStay;
                typeOfSport = "Athletics";
            } else if (typeOfGroup.equals("mixed")) {
                pricePerNight = numberOfStudents * 9.50;
                priceForWholeStay = pricePerNight * numberOfStay;
                typeOfSport = "Cycling";
            }
        } else if (season.equals("Summer")) {
            if (typeOfGroup.equals("boys")) {
                pricePerNight = numberOfStudents * 15.00;
                priceForWholeStay = pricePerNight * numberOfStay;
                typeOfSport = "Football";
            } else if (typeOfGroup.equals("girls")) {
                pricePerNight = numberOfStudents * 15.00;
                priceForWholeStay = pricePerNight * numberOfStay;
                typeOfSport = "Volleyball";
            } else if (typeOfGroup.equals("mixed")) {
                pricePerNight = numberOfStudents * 20.00;
                priceForWholeStay = pricePerNight * numberOfStay;
                typeOfSport = "Swimming";
            }
        }
        if (numberOfStudents >= 50) {
            priceForWholeStay = priceForWholeStay * 0.50;
        } else if (numberOfStudents >= 20 && numberOfStudents < 50) {
            priceForWholeStay = priceForWholeStay * 0.85;
        } else if (numberOfStudents >= 10 && numberOfStudents < 20) {
            priceForWholeStay = priceForWholeStay * 0.95;
        }
        System.out.printf("%s %.2f lv.", typeOfSport, priceForWholeStay);
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_3ConditionalStatementsAdvanced/_3MoreExercise/_07SchoolCamp.js
function schoolCamp(season, typeOfGroup, numberOfStudents, numberOfNights) {
    const PRICE_FOR_BOYS_OR_GIRLS_GROUP_AT_WINTER_BREAK = 9.60;
    const PRICE_FOR_BOYS_OR_GIRLS_GROUP_AT_SPRING_BREAK = 7.20;
    const PRICE_FOR_BOYS_OR_GIRLS_GROUP_AT_SUMMER_BREAK = 15.00;
    const PRICE_FOR_TOGETHER_GROUP_AT_WINTER_BREAK = 10.00;
    const PRICE_FOR_TOGETHER_GROUP_AT_SPRING_BREAK = 9.50;
    const PRICE_FOR_TOGETHER_GROUP_AT_SUMMER_BREAK = 20.00;

    const DISCOUNT_IF_STUDENTS_ARE_50_OR_MORE = 0.50;
    const DISCOUNT_IF_STUDENTS_ARE_20_TO_49 = 0.15;
    const DISCOUNT_IF_STUDENTS_ARE_10_TO_19 = 0.05;

    const SPORT_FOR_GIRLS_AT_WINTER_BREAK = "Gymnastics";
    const SPORT_FOR_GIRLS_AT_SPRING_BREAK = "Athletics";
    const SPORT_FOR_GIRLS_AT_SUMMER_BREAK = "Volleyball";
    const SPORT_FOR_BOYS_AT_WINTER_BREAK = "Judo";
    const SPORT_FOR_BOYS_AT_SPRING_BREAK = "Tennis";
    const SPORT_FOR_BOYS_AT_SUMMER_BREAK = "Football";
    const SPORT_FOR_TOGETHER_AT_WINTER_BREAK = "Ski";
    const SPORT_FOR_TOGETHER_AT_SPRING_BREAK = "Cycling";
    const SPORT_FOR_TOGETHER_AT_SUMMER_BREAK = "Swimming";

    let moneyForOneNightForGroup = 0.00;
    let sport;

    if (season === "Summer") {
        if (typeOfGroup === "boys") {
            moneyForOneNightForGroup = numberOfStudents * PRICE_FOR_BOYS_OR_GIRLS_GROUP_AT_SUMMER_BREAK;
            sport = SPORT_FOR_BOYS_AT_SUMMER_BREAK;
        } else if (typeOfGroup === "girls") {
            moneyForOneNightForGroup = numberOfStudents * PRICE_FOR_BOYS_OR_GIRLS_GROUP_AT_SUMMER_BREAK;
            sport = SPORT_FOR_GIRLS_AT_SUMMER_BREAK;
        } else if (typeOfGroup === "mixed") {
            moneyForOneNightForGroup = numberOfStudents * PRICE_FOR_TOGETHER_GROUP_AT_SUMMER_BREAK;
            sport = SPORT_FOR_TOGETHER_AT_SUMMER_BREAK;
        }
    } else if (season === "Winter") {
        if (typeOfGroup === "boys") {
            moneyForOneNightForGroup = numberOfStudents * PRICE_FOR_BOYS_OR_GIRLS_GROUP_AT_WINTER_BREAK;
            sport = SPORT_FOR_BOYS_AT_WINTER_BREAK;
        } else if (typeOfGroup === "girls") {
            moneyForOneNightForGroup = numberOfStudents * PRICE_FOR_BOYS_OR_GIRLS_GROUP_AT_WINTER_BREAK;
            sport = SPORT_FOR_GIRLS_AT_WINTER_BREAK;
        } else if (typeOfGroup === "mixed") {
            moneyForOneNightForGroup = numberOfStudents * PRICE_FOR_TOGETHER_GROUP_AT_WINTER_BREAK;
            sport = SPORT_FOR_TOGETHER_AT_WINTER_BREAK;
        }
    } else if (season === "Spring") {
        if (typeOfGroup === "boys") {
            moneyForOneNightForGroup = numberOfStudents * PRICE_FOR_BOYS_OR_GIRLS_GROUP_AT_SPRING_BREAK;
            sport = SPORT_FOR_BOYS_AT_SPRING_BREAK;
        } else if (typeOfGroup === "girls") {
            moneyForOneNightForGroup = numberOfStudents * PRICE_FOR_BOYS_OR_GIRLS_GROUP_AT_SPRING_BREAK;
            sport = SPORT_FOR_GIRLS_AT_SPRING_BREAK;
        } else if (typeOfGroup === "mixed") {
            moneyForOneNightForGroup = numberOfStudents * PRICE_FOR_TOGETHER_GROUP_AT_SPRING_BREAK;
            sport = SPORT_FOR_TOGETHER_AT_SPRING_BREAK;
        }
    }

    let totalMoneyForSchoolCamp = moneyForOneNightForGroup * numberOfNights;

    if (numberOfStudents >= 50) {
        totalMoneyForSchoolCamp = totalMoneyForSchoolCamp - (totalMoneyForSchoolCamp * DISCOUNT_IF_STUDENTS_ARE_50_OR_MORE);
    } else if (numberOfStudents >= 20) {
        totalMoneyForSchoolCamp = totalMoneyForSchoolCamp - (totalMoneyForSchoolCamp * DISCOUNT_IF_STUDENTS_ARE_20_TO_49);
    } else if (numberOfStudents >= 10) {
        totalMoneyForSchoolCamp = totalMoneyForSchoolCamp - (totalMoneyForSchoolCamp * DISCOUNT_IF_STUDENTS_ARE_10_TO_19);
    }

    console.log(`${sport} ${totalMoneyForSchoolCamp.toFixed(2)} lv.`);
}